tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -113,6 +114,43 @@ func newSession(credentialPath, credentialAccountID, region string) (*awssession
 	})
 }
 
+// retrievePrivateKey retrieves private key in format required for generating
+// public key using the privateKeyPath
+func (a *AwsProvider) retrievePrivateKey() (*rsa.PrivateKey, error) {
+	privateKeyBytes, err := ioutil.ReadFile(a.privateKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find private key from path: %v", a.privateKeyPath)
+	}
+
+	privateKeyBlock, _ := pem.Decode(privateKeyBytes)
+
```

Comment:
need to catch errors for privateKeyBlock here, i.e. if its nil

Reply:
we are not supporting this on azure through wni as per the ticket, as we just wanted to ensure public key authentication works and use it in wmco.

Reply:
done error check